proxy/pkg/zdmproxy/pscache.go

```diff
@@ -42,8 +59,8 @@ func (psc *PreparedStatementCache) Store(
 	psc.lock.Lock()
 	defer psc.lock.Unlock()
 
-	psc.cache[originPrepareIdStr] = NewPreparedData(originPreparedResult, targetPreparedResult, prepareRequestInfo)
-	psc.index[targetPrepareIdStr] = originPrepareIdStr
+	psc.cache.Add(originPrepareIdStr, NewPreparedData(originPreparedResult, targetPreparedResult, prepareRequestInfo))
+	psc.index.Add(targetPrepareIdStr, originPrepareIdStr)
```

Comment:
Hmm what happens if the limit is reached and the key that is selected to be evicted on one cache is different from the key that was selected on the other cache?

I'm not sure if bad behavior can happen if these two caches are out of sync or if it's fine...

If we do need both caches to be in sync then a potential alternative would be to provide an eviction callback to the `cache` that removes that key from the `index` cache. This way we could revert the `index` cache to a normal `map` (with our own locks) and perform the eviction ourselves.